src/edge/simulator.ts

```diff
@@ -91,85 +56,90 @@ export class Simulator {
         return null;
     }
 
-    private static async getLatestSimulatorVersion(): Promise<string | undefined> {
+    private static async checkCmdExist(cmd: string): Promise<boolean> {
         try {
-            const pipResponse = await request.get(Simulator.iotedgehubdevVersionUrl);
-            return JSON.parse(pipResponse).info.version;
+            await Executor.executeCMD(null, cmd, { shell: true }, "--version");
+            return true;
         } catch (error) {
-            return undefined;
+            return false;
         }
     }
 
-    private static async installSimulatorWithPip(force: boolean, message: string, outputChannel: vscode.OutputChannel = null): Promise<InstallReturn> {
-        // auto install only supported on windows. For linux/macOS ask user install manually.
-        if (os.platform() !== "win32") {
-            return InstallReturn.NotSupported;
-        }
-        let ret: InstallReturn = InstallReturn.Success;
-        if (await this.checkPipInstalled()) {
-            const install: vscode.MessageItem = { title: Constants.install };
-            const items: vscode.MessageItem[] = [ install ];
-            if (!force) {
-                items.push({ title: Constants.skipForNow });
-            }
+    private static adjustTerminalCommand(command: string): string {
+        return (Simulator.currentPlatform === "linux" || Simulator.currentPlatform === "darwin") ? `sudo ${command}` : command;
+    }
 
-            let input: vscode.MessageItem | undefined;
-            if (force) {
-                input = await vscode.window.showWarningMessage(message, { modal: true }, ...items);
-            } else {
-                input = await vscode.window.showWarningMessage(message, ...items);
-            }
+    private isInstalling: boolean = false;
+    private latestStandaloneSimulatorInfoJson: any;
+    private standaloneSimulatorExecutablePath: string;
```

Comment:
standalone may be confusing. "iotedgehubdev" is not standalone. Rename to simulatorExecutablePath?

Reply:
OK